app.rb

```diff
@@ -26,15 +28,18 @@ def add_person()
     when 1
       puts 'Has parent permission? [Y/N]'
       parent_permission = Input.new.parent_permission
-      student = Student.new(age, parent_permission, name)
+      type = 'Student'
+      student = Student.new(type, parent_permission, name, age)
+      @persons = []
       @persons.push(student)
     when 2
       puts 'Specialization'
       specialization = Input.new.input
-      teacher = Teacher.new(age, specialization, name)
+      type = 'Teacher'
+      teacher = Teacher.new(type, specialization, name, age)
       @persons.push(teacher)
     end
-    puts 'Person added successfully.'
+    write('Data Files/people.json', @persons)
```

Comment:
Instead of writing to files whenever an object is added to the lists, you should have a function that does that for all types of objects, and call that function when a person exits the application. Something like this: 
``` 
def save
    write('Data Files/people.json', @persons)
    write('Data Files/books.json', @books)
    write('Data Files/rentals.json', @rentals)
end
````
Then call it when a person exits the application.